bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusSenseHandler.java

```diff
@@ -121,38 +135,50 @@ private int getBatteryStatus(Appliance appliance) {
             if (!applianceStatusOptional.isPresent()) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
                         "Could not load data from API.");
-                return -1;
+                return null;
             }
 
             return applianceStatusOptional.get().getBatteryStatus();
         } catch (IOException e) {
             logger.debug("Could not load appliance status", e);
         }
-        return -1;
+        return null;
     }
 
     private @Nullable ApplianceData getApplianceData(Appliance appliance) {
+        // Dates are stripped of time part inside library
         Instant yesterday = Instant.now().minus(1, ChronoUnit.DAYS);
-        Instant today = Instant.now();
+        Instant tomorrow = Instant.now().plus(1, ChronoUnit.DAYS);
         OndusService service = getOndusService();
         if (service == null) {
             return null;
         }
         try {
-            BaseApplianceData applianceData = service.applianceData(appliance, yesterday, today).orElse(null);
-            if (applianceData.getType() != Appliance.TYPE) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                        "Thing is not a GROHE SENSE device.");
-                return null;
+            logger.debug("Fetching data for {} from {} to {}", thing.getUID(), yesterday, tomorrow);
+            BaseApplianceData applianceData = service.applianceData(appliance, yesterday, tomorrow).orElse(null);
+            if (applianceData != null) {
+                if (applianceData.getType() != Appliance.TYPE) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Thing is not a GROHE SENSE device.");
+                    return null;
+                }
+                return (ApplianceData) applianceData;
+            } else {
+                logger.debug("Could not load appliance data for {}", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Failed to find applicance data");
             }
-            return (ApplianceData) applianceData;
         } catch (IOException e) {
-            logger.debug("Could not load appliance data", e);
+            logger.debug("Could not load appliance data for {}", thing.getUID(), e);
         }
         return null;
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannels();
+            return;
```

Comment:
Well, openhab issues this command, so why not handle it and fetch latest values?

Reply:
Hard to track now since commit is gone, but I believe I was referring only to the line having the `return` statement as it was the last line in the method, if I remember correctly.

Reply:
Yes I understood that after while. And since the whole section was highlighted I thought you meant handling of Refresh command.

But anyways, the return statement is gone :)